### ISSUE TITLE
Remove RestoreSources from Tools.csproj

### ISSUE THREAD

**Issue by tmat** [dotnet/arcade#8144]
Remove https://github.com/dotnet/arcade/blob/main/eng/common/internal/Tools.csproj#L15-L23.
All repos should configure feeds in NuGet.config

**Comment by markwilkie** [comment]
cc/ @mmitche and @riarenas

**Comment by riarenas** [comment]
Agreed that we should do this. We just need to let partners know they'll need to add these two feeds to their NuGet.config

**Comment by riarenas** [comment]
Using an upstream source in the dotnet-core-internal-tooling feed allows us to add a compliant nuget.config to the project directory so that NuGet will use it. The remaining issue is that repos restore this project in different ways. MSBuild and Roslyn have a separate yaml step where they nuget restore the project while specifying the global nuget.config So this change as is would break them. Runtime restores using arcade scripts.

In order to avoid surprise official build breaks by removing the restore sources outright, I'm doing this in two steps:

Add the local nuget.config with the dotnet-core-internal-tooling feed: https://github.com/dotnet/arcade/pull/8309
provide guidance to the few repos that use this project so that their build pulls from the new nuget.config instead of using the one at the repo root.
Once the repos are using the new config a followup PR needs to be madeto delete the restoreSources from the csproj.

**Comment by riarenas** [comment]
First step is merged. Will wait until Arcade is promoted to let people know they need to restore the project using the new nuget.config

**Comment by riarenas** [comment]
Made a mistake in the original change. Fixing now and will add some additional validation to arcade-validation to prevent this kind of thing.

**Comment by riarenas** [comment]
Moving back to the backlog for a bit. We're pausing the sweep and clean while the JIT work is completed.
